src/components/accordions/PopularNovels/index.tsx

```diff
@@ -37,14 +60,31 @@ export default function PopularNovels() {
       <Stack direction={'row'} spacing={2}>
         <Button
           disabled={plugin === undefined}
-          onClick={fetchNovels}
+          onClick={() => fetchNovelsByIndex(1)}
           variant="contained"
         >
           Fetch
         </Button>
-        <Button disabled={plugin === undefined} variant="outlined">
+        <Button
+          disabled={currentIndex === 0}
+          variant="outlined"
+          onClick={() => fetchNovelsByIndex(currentIndex + 1)}
+        >
           Next Page
         </Button>
+        <ToggleButtonGroup
+          value={currentIndex}
+          exclusive
+          onChange={(event, value) => fetchNovelsByIndex(value)}
+        >
+          {Array.from({ length: maxIndex }, (_, index) => index + 1).map(
+            number => (
+              <ToggleButton key={`novel_page_index_${number}`} value={number}>
+                {number}
+              </ToggleButton>
+            ),
+          )}
+        </ToggleButtonGroup>
```

Comment:
I dont think we should use toggle for this case, 
try these steps: 
- go to page 2
- re-click on page 2
- press next page -> web go to page 1.
----
I think only list is enough.

Reply:
I just tested your scenario but last step I'm coming to page 3 which is well awaited

Reply:
dont fetch page 3 before you do the last step
step 1:
![image](https://github.com/LNReader/lnreader-plugins/assets/86464880/281ab623-b500-4cd2-9fae-be44a74ef8bc)
step 2:
![image](https://github.com/LNReader/lnreader-plugins/assets/86464880/0e8dfd2a-458d-4e21-a422-c5b5fbac911c)
step 3:
![image](https://github.com/LNReader/lnreader-plugins/assets/86464880/91e1b7fa-1111-4900-8921-f08c5ae9bd7a)

Reply:
What browser are you on? I tried firefox, edge and opera gx. I can't get back to step 2 or the item is deselected.

Reply:
I'm using chrome

Reply:
I've just tried on chrome, but I still don't reproduce. 
I may be stupid and I don't understand the step.
You could do a video discord to show me.

Reply:
I finally managed to reproduce on a different plugin than the one I'm on which I tested.